src/error.rs

```diff
@@ -4,6 +4,50 @@ use std::fmt;
 use Url;
 
 /// The Errors that may occur when processing a `Request`.
+///
+/// # Examples
+///
+/// ```
+/// #[macro_use]
+/// extern crate serde_derive;
+/// extern crate reqwest;
+///
+/// #[derive(Deserialize)]
+/// struct Simple {
+///    key: String
+/// }
+///
+/// // Response is not a json object conforming to the Simple struct
+/// fn run() -> Result<Simple, reqwest::Error> {
+///   reqwest::get("http://httpbin.org/ip")?.json()
+/// }
+/// fn main() {
```

Comment:
It'd be better if the doctests didn't actually make a network request. This can just be `# fn main() {}`.